src/databricks/labs/blueprint/paths.py

```diff
@@ -520,7 +520,23 @@ def resolve(self: P, strict: bool = False) -> P:
         if strict and not absolute.exists():
             msg = f"Path does not exist: {self}"
             raise FileNotFoundError(msg)
-        return absolute
+        # pylint: disable=protected-access
+        return absolute._normalize()
+
+    def _normalize(self: P) -> P:
+        if ".." not in self._path_parts:
+            return self
+        segments: list[str] = []
+        for part in self._path_parts:
+            if part == "..":
+                if segments:
+                    segments.pop()
+            elif part is None or part == '.':
+                continue
```

Comment:
I don't think this is needed: by the time we get here parts can neither be `None` nor `.`.